### PR TITLE
Always cancel the maxIterationTime and watchDog timers

### PR DESCRIPTION
**Pull request by jhford**
Even on success we want to cancel these timers.  Both calls are
idempotent, so it's no concern to call them twice in the case that the
race winner isn't the handler and the handler eventually completes and
cancels them a second time

### CODE REVIEW

**Comment by jhford** [comment]
What was happening is that these two timeouts were causing the timers to still live in the process, which kept anything which uses tc-lib-iterate from being able to exit unit tests without mocha's --exit flag.

File: src/iterate.js [hunk 1]
```diff
@@ -295,6 +263,9 @@ class Iterate extends events.EventEmitter {
     debug('stopped');
   }
 
+  cancelTimers() {
```
Comment (imbstack) [inline src/iterate.js]: Is this left over?

**Review by imbstack — APPROVED** [review]
Just a nit.

File: src/iterate.js (hunk 1 above)
Comment (jhford) [reply src/iterate.js]: yep!

**Pull request merged**